scripts/windows-msi.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Wix docs here: https://wixtoolset.org/documentation/tutorial/
+ * GUID Generator: http://www.guidgen.com/
+ * Author: Kanno
+ * Time: 10/5/2022
+ */
+const fs = require('fs-extra')
+const path = require('path')
+const execa = require('execa')
+const os = require('os')
+const msiInfo = require('./info.json')
+const pref_hooks = require('perf_hooks')
+
+const main = async () => {
+  const star = pref_hooks.performance.now()
+  const root = process.cwd()
+  const verPath = path.join(root, 'version.txt')
+  const packedPath = path.join(root, 'build')
+  try {
+    ;[verPath, packedPath].map((p) => fs.existsSync(p))
+    const ver = await fs.readFile(verPath, 'utf-8')
+    const winTars = ['build/grm-windows-32.tar.gz', 'build/grm-windows-64.tar.gz', 'build/grm-windows-arm64.tar.gz']
+    const args = ['-zxvf']
+    const windowsDir = []
+    await Promise.all(
+      winTars.map(async (p) => {
+        const out = 'windows' + '/' + p.split('/')[1].replace('.tar.gz', '')
+        windowsDir.push(out)
+        await fs.ensureDir(out)
+        execa('tar', [...args, p, '-C', out])
+      })
+    )
+
+    const info = Object.assign({}, msiInfo, { version: ver })
+
+    let msiTmpl = await fs.readFile(path.join(root, 'scripts', 'app.wsx.tmpl'), 'utf8')
+
+    // generator all wsx file.
+
+    Object.keys(info).forEach((c) => {
+      const reg = new RegExp(`{{.${c}}}`, 'g')
+      msiTmpl = msiTmpl.replace(reg, info[c])
+    })
+
+    await Promise.all(
+      windowsDir.map((dir) => {
+        const tpl = msiTmpl.replace('{{.buildSource}}', `${dir}/grm.exe`)
+        const out = path.join(root, dir, 'app.wsx')
+        fs.outputFile(out, tpl, 'utf8')
+      })
+    )
+
+    switch (os.platform()) {
+      case 'win32':
+        await Promise.all(
+          windowsDir.map(async (win) => {
+            await execa('candle.exe', ['-o', `${win}/app.wixobj`, `${win}/app.wsx`, '-ext', 'WixUtilExtension'])
+            await execa('light.exe', [
+              `${win}/app.wixobj`,
+              '-o',
+              `${win}/app.msi`,
+              '-ext',
+              'WixUIExtension',
+              '-ext',
+              'WixUtilExtension',
+            ])
+          })
+        )
+        break
+      case 'linux':
+        /**
+         * In linux system. Can use wixl to build msi installer.
+         * You can view docs here:http://manpages.ubuntu.com/manpages/bionic/man1/wixl.1.html
+         * I'm not sure darwin system can use wixl. If you found darwin system can use wixl too.
+         * You can modify this branch. Make it better :)
```

Comment:
It looks like this case is not used ? Do we need to package the msi in the CI?

Reply:
Currently, I done the `win32` part. In this weekend. I will optimization install process. And update Makefile. But build with `linux` or `darwin` system may be later.

Reply:
In my opinion it is best if ci can support it. Because if can't use ci. we should build msi file by we self.And upload to the release. From the notes at `windows-msi.js` you can know `wixl` can be work. But i'm afraid it will be deferred.

Reply:
It is make sense to me, other than that I'll verify locally that wix is packaged in linux and I'll update the linux section when this PR is done.

Reply:
Thanks. I'm  have not a linux  system. So i keep this branch in order to let later people see how to implement it. :)